addons/api/tests/unit/serializers/storage-bucket-test.js

```diff
@@ -57,6 +65,44 @@ module('Unit | Serializer | storage bucket', function (hooks) {
     assert.deepEqual(record.serialize(), expectedResult);
   });
 
+  test('it serializes a new dynamic aws plugin as expected', async function (assert) {
+    assert.expect(1);
+    const store = this.owner.lookup('service:store');
+    const record = store.createRecord('storage-bucket', {
+      compositeType: TYPE_STORAGE_BUCKET_PLUGIN_AWS_S3,
+      name: 'AWS',
+      description: 'this has an aws plugin',
+      bucket_name: 'bucketname',
+      bucket_prefix: 'bucketprefix',
+      worker_filter: 'workerfilter',
+      region: 'eu-west-1',
+      access_key_id: '',
+      secret_access_key: '',
+      secrets_hmac: 'hmac',
+      disable_credential_rotation: true,
+      role_arn: 'arn',
+      role_external_id: null,
+      role_session_name: null,
+      role_tags: [],
```

Comment:
Can we have these fields be non null so we can show that they get serialized correctly?

Reply:
Agreed with Zed's comment. We should make sure we serialize correctly those files with and without values.

Reply:
updated the attr type to use `object-as-array` transform as that is the expected backend format from the request.